department-of-veterans-affairs/vets-website
# 51172 - Choose a date error includes broken link to request flow

## Summary
This PR fixes the message displayed when an error occurs retrieving appointment slots and the clinic is not configured to accept appointment request. The appointment request link is also fixed and issue found in scenario when appointment request for clinic is enabled, user click link to request appointment, and then user clicks back button. Before fix, this scenario would result in a button with a spinning icon and an error logged to the console.

**NOTE:** The issue of 2 alert messages being displayed will be address in another ticket

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#51172

## Testing done
Unit and e2e testing

## How to test
To test this fix you will need to simulate a network error when retrieving appointment slots. Then can be done in Chrome developer tools by:

1. Scheduling an appointment like normal up to the 'Choose date and time' page
2. Go to 'Network' tab
3. Find and select the 'http://localhost:3000/vaos/v2/locations/983/clinics/455/slots?start=2023-01-01T00:00:00-06:00&end=2023-02-28T00:00:00-06:00' network request
4. Right click on this request and select 'Block request url'. This simulates the error
5. Refresh page and schedule the appointment again
6. You should see the error message when you get to the 'Choose date and time' page
7. Verify that the correct alert message is displayed for each scenario:
    - Clinic is configured for appointment request
    - Clinic is configured for appointment request and 'submit a request for a VA appointment' link is clicked
    - Clinic is configured for appointment request, 'submit a request for a VA appointment' link is clicked and back button is clicked on 'Choose and appointment day and time' page
    - Clinic is not configured for appointment request

**NOTE:** Clinic configuration for V2 is controlled via ./vaos/services/mocks/V2/scheduling_configuration.json. Find the location id, find the type of care, toggle the request.enabled value


## Screenshots
**Clinic configured to accept appointment request**
---
![localhost_3001_health-care_schedule-view-va-appointments_appointments_new-appointment_preferred-date](https://user-images.githubusercontent.com/3587066/212357536-7d507aaa-19dd-42b4-8141-0a161d5609f8.png)

**Clinic not configured to accept appointment request**
---
![localhost_3001_health-care_schedule-view-va-appointments_appointments_new-appointment_select-date](https://user-images.githubusercontent.com/3587066/212357643-49f5e085-cd05-4d7a-bc5f-f0173b881679.png)


## What areas of the site does it impact?
VAOS

## Acceptance criteria


## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
